src/main/java/mate/project/controller/BookController.java

```diff
@@ -27,16 +31,19 @@ public class BookController {
     private final BookService bookService;
 
     @GetMapping
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get a book by id", description = "Finding a book by specific id ")
     public BookDto getBookById(@PathVariable Long id) {
         return bookService.getBookById(id);
     }
 
     @PostMapping
+    @Operation(summary = "Create a new book", description = "Saving a new book to database")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto bookDto) {
         BookDto savedBook = bookService.save(bookDto);
```

Comment:
After saving a new book, it checks if the savedBook is null and throws an exception if it's null. However, it is generally considered better for the save method itself to throw an exception if it fails, rather than returning null.